students/rhamersky/Lesson_10/mailroom_part6.py

```diff
@@ -0,0 +1,392 @@
+    def user_filter_input():
+        '''
+        Try and except block used on User's option choice to check for integers.
+        :return: Returns filter data to the main function.
+        '''
+        while True:
+            try:
+                str_user_input = input(str_name + ", please enter max to filter for amounts greater than or equal "
+                                                  "to $2,500. \n or min to filter for "
+                                                  "amounts less than $2,500. ").capitalize()
+            except ValueError:
+                print(str_name + ", not a valid option. \n")
+            # Verifies that the data type is part of the string class and that the user put in the correct input.
+            else:
+                if str_user_input.__class__ == str().__class__ and (str_user_input == "Max"
+                                                                    or str_user_input == "Min"):
+                    break
+                else:
+                    print("Integer selected is not a valid option. "
+                              "Please, enter max or min. \n")
+        return str_user_input
+
+########### END OF MENU CLASS###############
+
+# Methods defined in the files class are related to opening and renaming files.
+# Modifications to the methods are needed if the files are not for a task priority list.
+
+
+class Files:
+    @staticmethod
+    def write_email(donors):
+        '''
+        Writes emails to the donors and saves them to text files.
+        :param donors: Dictionary that stores all the donor's donation information.
+        :return:
+        '''
+        for donor_name in donors:
+            # Open text file and assign a mode to the file.
+            new_file = open(donor_name + ".txt", mode="w")
+            new_file.write(Files.donor_email(donor_name))
+            new_file.close()
+            print(donor_name + " saved to "+donor_name+".txt.")
+
+    @staticmethod
+    def donor_email(donor):
+        '''
+        Formats the email with donor's name and total lifetime donations.
+        :param donor: Dictionary that stores all the donor's donation information.
+        :param dictionary: A dictionary that stores the donor as the 'key' and the lifetime donation as the 'value'.
+        :return: The concatenation how the three strings.
+        '''
+        str_salutation = "Dear {},".format(donor)
+        str_body = "\n\n"+"Thank you for generosity! " \
+                   "\n\n"+"This donation will be put to great use!" \
+                   "\n\n"+"Your current lifetime donation is "+"${:,.02f}".format(dic_donors[donor][0]) + "."
+        str_valediction = "\n\n"+"Sincerely," \
+                          "\n\n"+"The Team"
+        thankyou_message = str_salutation+str_body+str_valediction
+        return thankyou_message
+
+############# END OF FILE CLASS #################
+
+# The Data class is where the data get manipulated.
+
+
+class Data:
```

Comment:
In general, it's better to use class/function/variable names that are more specific to their use case in your program e.g. `DonorsData`